### PR TITLE
Fix #4974: Update GAVIN logic to match algorithm in last version of the paper

### LINKED ISSUE #4974: Update GAVIN logic to match algorithm in last version of the paper
GAVIN logic no longer reflects the algorithm in the paper. Update it.

### PR DESCRIPTION
**Pull request by joerivandervelde**
Fixed GAVIN logic, now matches the algorithm used in latest paper version
Refactored the classification function, now used in downstream dependencies, removing duplicated code
Cleanup of Impact class, no longer duplicated
Fixed typo of 'pathogenic'
Added comments and explanations

### CODE REVIEW

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/gavin/GavinAlgorithm.java [hunk 1]
```diff
@@ -12,115 +16,126 @@
...
+				category = geneToEntry.get(gene).category;
 			}
 		}
 
 		// CADD score based classification, calibrated
-		if (caddScaled != null)
+		if(caddScaled != null)
 		{
-			if ((category.equals(Category.C1) || category.equals(Category.C2)))
+			if((category.equals(Category.C1) || category.equals(Category.C2)))
```
Comment (fdlk) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/gavin/GavinAlgorithm.java]: Make this a switch statement, it'll look cleaner

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/gavin/GavinAlgorithm.java [hunk 2]
```diff
@@ -12,115 +16,126 @@
...
+		{
+			return new Judgment(Judgment.Classification.Benign, Method.calibrated, gene, "Variant MAF of " + exacMAF + " is greater than "+pathoMAFThreshold+".");
+		}
+
+		String mafReason = "the variant MAF of " + exacMAF + " is less than a MAF of "+ pathoMAFThreshold + ".";
+
+		// Impact based classification, calibrated
+		if(impact != null)
+		{
+			if(category.equals(Category.I1) && impact.equals(Impact.HIGH))
```
Comment (fdlk) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/gavin/GavinAlgorithm.java]: You can compare enums using ==, need not use .equals

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/gavin/GavinAlgorithm.java [hunk 3]
```diff
@@ -12,115 +16,126 @@
...
+					return new Judgment(Judgment.Classification.Pathogenic,  Method.calibrated, gene, "Variant CADD score of " + caddScaled + " is greater than "+spec95thPerCADDThreshold+" for this gene.");
 				}
-				else if (caddScaled < spec95thPerCADDThreshold)
+				else if(caddScaled < sens95thPerCADDThreshold)
 				{
-					return new Judgment(Judgment.Classification.Benign, Judgment.Method.calibrated,
-							"Variant CADD score of " + caddScaled + " is lesser than the 95% sensitivity threshold of "
-									+ spec95thPerCADDThreshold + " for this gene, although " + mafReason);
+					return new Judgment(Judgment.Classification.Benign,  Method.calibrated, gene, "Variant CADD score of " + caddScaled + " is less than "+sens95thPerCADDThreshold+" for this gene.");
 				}
```
Comment (fdlk) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/gavin/GavinAlgorithm.java]: else?

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/gavin/GavinAlgorithm.java [hunk 4]
```diff
@@ -12,115 +16,134 @@
...
+		if(geneToEntry == null)
 		{
-			return new Judgment(Judgment.Classification.Benign, Judgment.Method.calibrated, "Variant MAF of " + exacMAF
-					+ " is greater than the pathogenic 95th percentile MAF of " + pathoMAFThreshold + ".");
+			//get data from entity for the annotator
+			pathoMAFThreshold = annotationSourceEntity.getDouble(PATHOMAFTHRESHOLD) * extraSensitivityFactor * 2;
+			meanPathogenicCADDScore = annotationSourceEntity.getDouble(MEANPATHOGENICCADDSCORE) - extraSensitivityFactor;
+			meanPopulationCADDScore = annotationSourceEntity.getDouble(MEANPATHOGENICCADDSCORE) - extraSensitivityFactor;
+			spec95thPerCADDThreshold = annotationSourceEntity.getDouble(SPEC95THPERCADDTHRESHOLD) - extraSensitivityFactor;
+			sens95thPerCADDThreshold = annotationSourceEntity.getDouble(SPEC95THPERCADDTHRESHOLD) - extraSensitivityFactor;
```
Comment (Mark-de-Haan) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/gavin/GavinAlgorithm.java]: These constants are wrong.
Requires null check

**Comment by jjettenn** [comment]
Do you still plan to fix it?

**Comment by jjettenn** [comment]
We will create e PR from MOLGENIS 1.21 to master

**Comment by jjettenn** [comment]
Closed because this PR is not containing the extra bug fixes that where added for MOLGENIS 1.21

**Pull request closed without merging**